godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/array/core/AABBArray.kt

```diff
@@ -0,0 +1,35 @@
+package godot.core
+
+import godot.gdnative.godot_array
+import kotlinx.cinterop.COpaquePointer
+import kotlinx.cinterop.CValue
+
+@ExperimentalUnsignedTypes
+class AABBArray : CoreArray<AABB> {
+    constructor() : super()
+    constructor(other: AABBArray) : super(other)
+    internal constructor(native: CValue<godot_array>) : super(native)
+    internal constructor(mem: COpaquePointer) : super(mem)
+
+    override fun getCore(value: Variant): AABB = value.asAABB()
+    override fun getCoreArray(value: CValue<godot_array>) =
+        AABBArray(value)
+}
+
+@ExperimentalUnsignedTypes
+fun AABBArrayOf(vararg elements: AABB) = AABBArray().also {
+    for (arg in elements) {
+        it.append(arg)
+    }
+}
+
+@ExperimentalUnsignedTypes
+fun Iterable<AABB>.toVariantArray() = AABBArray().also{
+    for (arg in this) {
+        it.append(arg)
+    }
+}
+
+@ExperimentalUnsignedTypes
+fun AABBArray(iter: Iterable<AABB>) = iter.toVariantArray()
```

Comment:
Same with this function. I really like this idea of you. But i would ad the kDoc here as well as this "constructor" will not perform well, but it's needed.